dali/kernels/tensor_view.h

```diff
@@ -267,12 +316,16 @@ struct TensorListView<Backend, DataType, DynamicDimensions>
                  std::vector<ptrdiff_t> &&offsets)
       : Base(data, std::move(shape), std::move(offsets)) {}
 
-  template <int other_sample_ndim>
-  TensorListView(const TensorListView<Backend, DataType, other_sample_ndim> &other)
-      : Base(other.data, other.shape, other.offsets) {}
-  template <int other_sample_ndim>
-  TensorListView(TensorListView<Backend, DataType, other_sample_ndim> &&other)
-      : Base(other.data, std::move(other.shape), std::move(other.offsets)) {
+  template <int other_sample_ndim, typename U>
+  TensorListView(const TensorListView<Backend, U, other_sample_ndim> &other)
+  : Base(other.data, other.shape, other.offsets) {
```

Comment:
This is one of three ways to indent initializer list that I can (sadly) find in our code base, it is not compatible with style-guide https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists and .clang-format. 
And it is inconsistent in this file.

I would suggest to stick to one thing with formatting, and advise to use clang-format for the new files.